### PR TITLE
Make sure posterizer link make one request only

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by norkans7** [comment]
This is the proper way to make sure we include one posterizer form on any page

**Comment by nicpottier** [comment]
So can you tell us a bit more what's happening here. I gather the main thing is we do NOT wire formax on requests that are formax requests is that right? Have you tested this on pages that have inline forms that are formax (triggers, org) to make sure nothing is broken?

**Comment by norkans7** [comment]
@nicpottier sorry I had missed this comment.
formax submits the form using https://github.com/nyaruka/rapidpro/blob/master/static/coffee/formax.coffee#L84 to the URL of the formax section so that is the reason we do not want the requests for the posterizer form 
I tested the pages and everything looks right

**Comment by rowanseymour** [comment]
@norkans7 can you point me to example in RP where we're currently doing the wrong thing? Just want to get a better understanding of what the problem is, and I'm nervous to make a change that would effect so many forms and in a way that wouldn't be picked up by unit tests.

**Comment by norkans7** [comment]
The issue was that we could add many voice caller for a channel https://github.com/rapidpro/rapidpro/issues/1094

1) I added a backend validation https://github.com/nyaruka/rapidpro/pull/2601
2) WE found out the issue was the formax making multiple requests this PR addresses that

**Comment by rowanseymour** [comment]
@norkans7 to clarify, it's hitting the _Enable Voice_ button here that can lead to 2 requests being made? I'm trying to replicate that but can't...

Captura de Pantalla 2019-12-10 a la(s) 15 23 41

**Comment by norkans7** [comment]
Are you checking the browser console?

**Comment by rowanseymour** [comment]
Yeah and now I see it.. there isn't just 2 requests.. there are 15 requests to create_caller like..

```
&channel=11015
```
```
&channel=11015&channel=11015
```
```
...
```
```
&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015&channel=11015
```

The first 14 are cancelled and the 15th succeeds. I see 19 posterizer forms in the page source code.. any idea why it's making 15 requests?

**Comment by norkans7** [comment]
That is the issue you reproduced and this PR makes sure only one request is made and submitted

File: templates/frame.haml [hunk 1]
```diff
@@ -385,9 +385,10 @@
 
     <script type="text/javascript" src="{{ STATIC_URL }}js/libs/jquery.url.js"></script>
 
-    <form id="posterizer" method="post">
-      {% csrf_token %}
-    </form>
+    - if not request.META.HTTP_X_FORMAX or formax
```
Comment (rowanseymour) [inline templates/frame.haml]: Does a formax request ever use the frame template? Wondering if this is needed

Comment (norkans7) [reply templates/frame.haml]: I think it is needed so we are able to posterize

Comment (rowanseymour) [reply templates/frame.haml]: This is saying "if we're not a formax request, add posterizer stuff".. but I don't think formax requests use frame.haml. I just checked this out and tried it locally and it seems it works fine without the check

File: templates/frame.haml [hunk 2]
```diff
@@ -435,7 +436,7 @@
           timeout:300
         });
 
-
+        {% if not request.META.HTTP_X_FORMAX or formax %}
```
Comment (rowanseymour) [inline templates/frame.haml]: as above

File: templates/frame.haml (hunk 1 above)
Comment (norkans7) [reply templates/frame.haml]: both the context variable `formax` and `request.META.HTTP_X_FORMAX` are on the view 'orgs_org_home' not on the request that requests the sections part

I will check on that again and see

**Comment by rowanseymour** [comment]
@norkans7 thanks, can't believe we didn't run into problems with this before but merged now and shipping shortly

**Pull request merged**